File: PHP/8php 2/minmax.php
<?php
/*
|--------------------------------------------------------------------------
| Find Maximum and Minimum Values of a List
|--------------------------------------------------------------------------
|
*/

function maximum($array) {
   return max($array);
}
function minimum($array) {
   return min($array);
}


use function max as maximum;
use function min as minimum;


function maximum($array) {
   return array_reduce($array, function ($acc, $cur) { return $cur > $acc ? $cur : $acc; }, $array[0]);
}
 function minimum($array) {
   return array_reduce($array, function ($acc, $cur) { return $cur < $acc ? $cur : $acc; }, $array[0]);
}


function maximum($array) {
   sort($array);
   $array = array_reverse($array);
   return $array[0];
}
function minimum($array) {
   sort($array);
   return $array[0];
}


print_r(maximum(array(1, 2, 3, 4, 5)). ' ' . 5 . '
');
print_r(minimum(array(1, 2, 3, 4, 5)). ' ' . 1 . '
');

File: Python/8py 2/minmax.py
"""
|--------------------------------------------------------------------------
| Find Maximum and Minimum Values of a List
|--------------------------------------------------------------------------
|
"""

def minimum(arr):
   return min(arr)
def maximum(arr):
   return max(arr)


minimum,maximum = min,max


def min(arr):
   return sorted(arr)[0]
def max(arr):
   return sorted(arr)[-1]


from functools import reduce
def min(arr):
   return reduce(lambda x, y: x if x < y else y, arr)
def max(arr):
   return reduce(lambda x, y: x if x > y else y, arr)


def min(arr):
   return vars(__builtins__)['min'](arr)
def max(arr):
   return vars(__builtins__)['max'](arr)


print(minimum([-52, 56, 30, 29, -54, 0, -110]), -110)
print(minimum([42, 54, 65, 87, 0]), 0)
print(maximum([-52, 56, 30, 29, -54, 0, -110]), 56)
print(maximum([4,6,2,1,9,63,-134,566]), 566)